### PR TITLE
Added a rule for removing artifacts by regex

### LINKED ISSUE #111: Possibility to use Regex
It would be interesting if there was the possibility of using regex when configuring a rule in the artifactory-cleanup.yaml file, as in the example below:
       
```
 - rule: IncludeFilename
  masks: 
   - "*/production-[0-9]+"
   - "*.develop-[0-9]+"
```

That way I believe you would have more control over what you want to look for in a repository.

### PR DESCRIPTION
**Pull request by Hardar3**
Allows using `DeleteByRegex` for removing artifacts by redex
Resolves  #111

### CODE REVIEW

File: artifactory_cleanup/rules/delete.py [hunk 1]
```diff
@@ -102,3 +103,23 @@ def filter(self, artifacts):
...
+    """
+    Remove artifacts by regex pattern.
+    """
+
+    def __init__(self, regex_pattern):
+        self.regex_pattern = rf"{regex_pattern}"
+
+    def aql_add_filter(self, filters):
+        print(filters)
+        print(type(filters))
```
Comment (allburov) [inline artifactory_cleanup/rules/delete.py:118]: If you wanna have them for debug - please add some message, like "Here's filters that we get"

File: artifactory_cleanup/rules/delete.py [hunk 2]
```diff
@@ -102,3 +103,23 @@ def filter(self, artifacts):
...
+        return filters
+
+    def filter(self, artifacts: ArtifactsList) -> ArtifactsList:
+        for  val in artifacts[:]:
```
Comment (allburov) [inline artifactory_cleanup/rules/delete.py:122]: We can name it explicitly - it's `artifact`, not just some value

Comment (allburov) [reply artifactory_cleanup/rules/delete.py:122]: in that case `artifact['name']` means something, but with `val['name']` it's not obviousl

File: artifactory_cleanup/rules/delete.py [hunk 3]
```diff
@@ -102,3 +103,23 @@ def filter(self, artifacts):
         repositories = utils.build_repositories(artifacts)
         folders = utils.get_empty_folders(repositories)
         return folders
+
+
+class DeleteByRegex(Rule):
```
Comment (allburov) [inline artifactory_cleanup/rules/delete.py:108]: Could you add some documentation for the rule in readme.md please? So we one can see how to use it

File: artifactory_cleanup/rules/delete.py [hunk 4]
```diff
@@ -102,3 +103,23 @@ def filter(self, artifacts):
...
+    """
+    Remove artifacts by regex pattern.
```
Comment (allburov) [inline artifactory_cleanup/rules/delete.py:110]: Does it affect only `name` or the whole `path`? 
I think we could explicitly name it `DeleteByRegexpName` to highlight that it works only in `name` attribute

**Review by allburov — COMMENTED** [review]
Hi! Looks good, but I've added few comments how we can improve it

**Comment by Hardar3** [comment]
@allburov Hello! I fix your comments
Thank you for feedback

**Comment by felipecrs** [comment]
Am I right that I can't combine `DeleteOlderThan` and `DeleteByRegexpName` to ensure I only delete artifacts that match both?

**Comment by allburov** [comment]
why not?

**Comment by felipecrs** [comment]
It is not clear what is the different between Delete* rules and Include* rules. I suspected that the Delete* rules would always delete any artifacts that matches it, regardless of other Delete* rules. Unless they are filtered by Include* rules first.

**Pull request merged**